storage/include/storage/storage_metrics.h

```diff
@@ -64,8 +64,8 @@ class InMemoryStorageMetrics {
  */
 class RocksDbStorageMetrics {
```

Comment:
I think we need to think about the `last_metrics_update_` variable too as it might be changed concurrently.

Reply:
done

Reply:
Does the mutex also protect the atomic counters? If not, maybe we can also use an atomic for the last update time, because that mutex will be locked often. Please correct me if I am wrong, I am using my phone and I don’t see the code really well.

Reply:
The mutex protects only the `last_metrics_update_`. We cannot use atomic because it supports only primitive types.
This mutex will be taken only once in a while when we update the aggregator (timer depended)

Reply:
Don’t we acquire it every time on tryToUpdateMetrics()? That is called on every DB operation, right? I am still on my phone, sorry if I miss something.

Maybe we can use another type for time that can be atomic, just a thought.

Reply:
you right, I will fix it

Reply:
done. 
I've changed the update to be operated based instead of time based. And made it atomic.